### DISCUSSION TITLE
Staticcheck doesn't display live problems

### DISCUSSION THREAD

**Discussion by ousloob** [golang/vscode-go#2067] [Help] [answered]
Hello,
I installed staticcheck but it doesn't tell me any problems on my vscode.
I configured my linter to use staticcheck and I looked everywhere on internet it still doesn't work and its drive me crazy.

Here is a part of my settings. json file:

`{
  "go.formatTool": "goimports",
  "go.useLanguageServer": true,
  "go.lintTool": "staticcheck",
  "go.lintFlags": [],
  "go.testFlags": ["-v"],
  "go.toolsManagement.autoUpdate": true,
  "editor.codeActionsOnSave": {
    "source.fixAll.eslint": true
  }
`

**Comment by butuzov** [comment]
Just in case this is a duplicate of user request on StackOverflow - [`How can I configure the 'staticcheck' linter in Visual Studio Code?`](https://stackoverflow.com/questions/71101439/how-can-i-configure-the-staticcheck-linter-in-visual-studio-code)

1) you dont need `"go.lintTool": "staticcheck"` if you using `staticcheck`, its default linter for `vscode-go`
2) also (with language server enabled) - you  need to have 

```
"gopls": { 
    "ui.diagnostic.staticcheck": true 
}, 
"go.languageServerExperimentalFeatures": {
        "diagnostics": true
 },
```
P/S/
This these settings in place I am able to see errors in code, I am not able to see problems (in panel/sidebar).

**Reply by ousloob** [reply]
Hey, Yes same post same person. :) 
Tries this one too, still no errors neither in code neither in the sidebar

**Comment by hyangah** [comment, marked as answer]
We need more details on your setup (i.e. is there a lint error that you expected? did you open the file? what is the code layout?). But if what you want is to run lint for all packages in the workspace, can you try to run "Go: Lint Workspace" (not with gopls's staticcheck, but with the default settings)? Or set `"go.lintOnSave": "workspace"` and save a file to trigger lint.
 
I heard gopls does not report lint errors for unopened files/packages - that's too expensive.
In the legacy lint tool mode (not gopls mode), the default linting works on the current 'package'.

Another downside of `staticcheck` option offered by `gopls` is currently there is no configuration option.

**Reply by ousloob** [reply]
Yes I was expecting an [ST1000 ](https://staticcheck.io/docs/checks/#ST1000) error.
To get the error I have to add this check in my _settings.json_
```json
"ui.diagnostic.analyses": {
  "ST1000": true,
},
```
And yes I have to go to the specific file to get the error.
But I already saw a case where multiple errors appears in all the workspace. Your solution with `"go.lintOnSave"` explain why.

Now you say there is no configuration option of **staticchec** offered by gopls, it's half true. Cause I can only add the checks I need one bye one like this:
```json
"ui.diagnostic.analyses": {
  "ST1000": true,
  "ST1003": true,
  "ST1016": true,
},
```

**Reply by hyangah** [reply]
The gopls issue about applying staticcheck's standard configuration is https://github.com/golang/go/issues/36373
Compilation errors or critical errors appear in all workspace files. Most lint errors don't IIUC.
The default staticcheck enabled (recognized by gopls) are in https://github.com/golang/tools/blob/master/gopls/internal/hooks/analysis.go and yes, you can choose to disable the checks added by default, one by one.
